### PR TITLE
loading json

### PR DESCRIPTION
**Pull request by zafrani**
Merge after fragments

### CODE REVIEW

**Review by Zsteven44 — APPROVED** [review]

File: app/src/main/java/zafrani/com/pubgapp/activities/DrawerActivity.java [hunk 1]
```diff
@@ -73,9 +77,50 @@ public boolean onNavigationItemSelected(@NonNull final MenuItem item) {
...
+        Fragment fragment = getFragmentManager().findFragmentByTag(ItemFragment.TAG);
+        if (fragment == null) {
+            showFragment(new ItemFragment());
+        } else {
+            showFragment(fragment);
+        }
+    }
+
+    private void showFragment(@NonNull final Fragment fragment) {
+        final FragmentTransaction fragTransaction = getFragmentManager().beginTransaction();
```
Comment (esanz91) [inline app/src/main/java/zafrani/com/pubgapp/activities/DrawerActivity.java:119]: maybe just:

```
FragmentManager manager = getFragmentManager();

manager.replace(R.id.activity_drawer_content, fragment)
    .commit();
manager.executePendingTransactions();
```

File: app/src/main/java/zafrani/com/pubgapp/adapters/BaseFragmentAdapter.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        super(fm);
+        this.context = context;
+        this.fragments = fragments;
+        this.rangeOffset = new RangeOffset(0, fragments.size());
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        final int offsetPosition = getPositionWithOffset(position);
+        Log.e(getClass().getSimpleName(), "getPageTitle: " + position + " | " + offsetPosition);
```
Comment (esanz91) [inline app/src/main/java/zafrani/com/pubgapp/adapters/BaseFragmentAdapter.java:34]: did this slip through? remove log if not needed

File: app/src/main/java/zafrani/com/pubgapp/adapters/BaseFragmentAdapter.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v13.app.FragmentStatePagerAdapter;
+import android.util.Log;
+
+import java.util.List;
+
+import zafrani.com.pubgapp.fragments.BaseFragment;
+
+public class BaseFragmentAdapter extends FragmentStatePagerAdapter {
```
Comment (esanz91) [inline app/src/main/java/zafrani/com/pubgapp/adapters/BaseFragmentAdapter.java:14]: Seems like you're not even using this class. Please delete it. :)

File: app/src/main/java/zafrani/com/pubgapp/fragments/BaseFragment.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package zafrani.com.pubgapp.fragments;
+
+
+import android.app.Fragment;
+
+public abstract class BaseFragment extends Fragment {
```
Comment (esanz91) [inline app/src/main/java/zafrani/com/pubgapp/fragments/BaseFragment.java:6]: what's the point of this?

File: app/src/main/java/zafrani/com/pubgapp/fragments/ItemFragment.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        return inflater.inflate(R.layout.fragment_item, container, false);
+
+    }
+
+    @Override
+    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        try {
+            Items items = FileUtil.getItems(getActivity());
+            Log.e(getClass()
```
Comment (esanz91) [inline app/src/main/java/zafrani/com/pubgapp/fragments/ItemFragment.java:34]: delete logs, if not needed

File: app/src/main/java/zafrani/com/pubgapp/utils/FileUtil.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+public class FileUtil {
+
+    @Nullable
+    public static Items getItems(@NonNull final Activity activity) throws IOException {
+        final InputStream inputStream = activity.getAssets().open("items.json");
+
+        if (inputStream != null) {
+            final Gson gson = new Gson();
+            final BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+            return gson.fromJson(reader, Items.class);
```
Comment (esanz91) [inline app/src/main/java/zafrani/com/pubgapp/utils/FileUtil.java:25]: 🎉

File: app/src/main/res/layout/fragment_item.xml [hunk 7]
```diff
@@ -3,10 +3,10 @@
     android:orientation="vertical"
     android:layout_width="match_parent"
     android:layout_height="match_parent">
-
     <TextView
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Gear Fragment"/>
+        android:textColor="#fff"
```
Comment (esanz91) [inline app/src/main/res/layout/fragment_item.xml:9]: `android:@color/white`

File: app/src/main/res/layout/fragment_map.xml [hunk 8]
```diff
@@ -6,6 +6,7 @@
     <TextView
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
+        android:textColor="#fff"
```
Comment (esanz91) [inline app/src/main/res/layout/fragment_map.xml:9]: `android:@color/white`

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1,3 +1,6 @@
 <resources>
     <string name="app_name">pubgapp</string>
+    <string name="map_tab_title">Maps</string>
```
Comment (esanz91) [inline app/src/main/res/values/strings.xml:3]: delete these if we're not using them, but I think we should follow this pattern:

`${type}_${description}` i.e., `title_map`, `title_tab`

**Pull request merged**